### PR TITLE
docs: extend documentation

### LINKED ISSUE #206: OpenSSF best practices

### PR DESCRIPTION
**Pull request by ramizpolic**
## Overview
This PR adds/fixes helm docs and adds OpenSSF Best Practice badge to the repo readme.

Fixes #206 #18

### CODE REVIEW

**Review by akijakya — DISMISSED** [review]
Nicely done, thanks!

**Review by akijakya — APPROVED** [review]

**Review by akijakya — APPROVED** [review]
Added some comments about obsolete stuff that can be fixed in a next step, otherwise LGTM, thanks!

**Comment by akijakya** [comment]
I just noticed a thing: there is a `kubeVersion` badge defined the gotmpl file, but since this field is missing from the Chart.yaml, it won't be added to the Readme (it looks for `kubeVersion` in the Chart.yaml instead of values.yaml).

Another thing is that updating the chart readme could be added to the pre-commit hooks, so it would refresh automatically when a something is changed around the chart, what do you think?

File: deploy/charts/vault-secrets-webhook/values.yaml [hunk 1]
```diff
@@ -2,157 +2,212 @@
...
 
+# -- Node labels for pod assignment.
+# Check: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector
 nodeSelector: {}
 
+# -- List of node tolerations for the pods.
+# Check: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
 tolerations: []
 
+# Node affinity settings for the pods.
```
Comment (akijakya) [inline deploy/charts/vault-secrets-webhook/values.yaml:164]: The double dashes are missing from here

File: deploy/charts/vault-secrets-webhook/README.md.gotmpl [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+{{ template "before-config" . }}
+{{- end -}}
+
+{{- define "chart.base" -}}
+{{ template "chart.baseHead" . }}
+
+{{ template "chart.valuesSectionHtml" . }}
+
+{{ template "after-config" . }}
+{{- end -}}
```
Comment (akijakya) [inline deploy/charts/vault-secrets-webhook/README.md.gotmpl:176]: Are these definitions necessary here?

File: deploy/charts/vault-secrets-webhook/README.md.gotmpl [hunk 3]
```diff
@@ -0,0 +1,190 @@
+{{- define "chart.kubeVersionBadge" -}}
+{{- if .KubeVersion -}}![kube version: {{ .KubeVersion }}](https://img.shields.io/badge/kube%20version-{{ .KubeVersion | replace "-" "--" }}-informational?style=flat-square){{- end -}}
```
Comment (akijakya) [inline deploy/charts/vault-secrets-webhook/README.md.gotmpl:2]: I'd like it better if the text said `kubernetes version:`, also as I mentioned in a previous comment, it is currently not added to the final Readme as it looks for a `kubeVersion` field in the Chart.yaml, which is currently missing. Is this badge necessary btw?

**Review by akijakya — CHANGES_REQUESTED** [review]
As I fiddled with it, I found some issues I previously overlooked...

File: deploy/charts/vault-secrets-webhook/README.md.gotmpl (hunk 2 above)
Comment (ramizpolic) [reply deploy/charts/vault-secrets-webhook/README.md.gotmpl]: No need for these templates at all (this enables full helm docs reconfiguration). Replaced with only data injected before and after values table.

File: deploy/charts/vault-secrets-webhook/README.md.gotmpl (hunk 3 above)
Comment (ramizpolic) [reply deploy/charts/vault-secrets-webhook/README.md.gotmpl]: Removed, not needed

**Review by akijakya — APPROVED** [review]
LGTM, thanks!

**Pull request merged**